clouddriver-ecs/src/main/java/com/netflix/spinnaker/clouddriver/ecs/services/ContainerInformationService.java

```diff
@@ -95,18 +101,41 @@ public ContainerInformationService(
 
     // Task-based health
     if (task != null) {
+      String taskDefinitionCacheKey =
+          Keys.getTaskDefinitionKey(accountName, region, service.getTaskDefinition());
+      TaskDefinition taskDefinition = taskDefinitionCacheClient.get(taskDefinitionCacheKey);
+
+      boolean hasHealthCheck = false;
+
+      for (ContainerDefinition containerDefinition : taskDefinition.getContainerDefinitions()) {
+        if (containerDefinition.getHealthCheck() != null
+            && containerDefinition.getHealthCheck().getCommand() != null) {
+          hasHealthCheck = true;
+          break;
+        }
+      }
+
       Map<String, Object> taskPlatformHealth = new HashMap<>();
       taskPlatformHealth.put("instanceId", taskId);
       taskPlatformHealth.put("type", "ecs");
       taskPlatformHealth.put("healthClass", "platform");
-      taskPlatformHealth.put("state", toPlatformHealthState(task.getLastStatus()));
+      taskPlatformHealth.put(
+          "state",
+          toPlatformHealthState(task.getLastStatus(), task.getHealthStatus(), hasHealthCheck));
       healthMetrics.add(taskPlatformHealth);
     }
 
     return healthMetrics;
   }
 
-  private String toPlatformHealthState(String ecsTaskStatus) {
+  private String toPlatformHealthState(
+      String ecsTaskStatus, String ecsTaskHealthStatus, boolean hasHealthCheck) {
+    if (hasHealthCheck && "UNKNOWN".equalsIgnoreCase(ecsTaskHealthStatus)) {
+      return "Starting";
+    } else if ("UNHEALTHY".equals(ecsTaskHealthStatus)) {
+      return "Down";
+    }
```

Comment:
why equalsIgnoreCase on one but not the other check against ecsTaskHealthStatus?

Reply:
updated both to equals (as that's what we're expecting)